qualang_tools/simulator_tools.py

```diff
@@ -1,7 +1,8 @@
+import numpy as np
 from qm import ControllerConnection, InterOpxChannel
 
 
-def create_simulator_controller_connections(n_controllers, n_connections=12):
+def create_simulator_controller_connections(n_controllers, n_connections=12, old_method=False, n_actual_controllers=None):
```

Comment:
If I understand correctly, n_controllers=6 and n_actual_controller=4 would mean that you want to "build" only a subset of the 6 controllers and it will be from 1 to 4, is it correct?
What is the use-case for this?

Reply:
Yes, exactly.
It came from TII, which has a 9 OPX cluster connected but then opened a 3 OPX cluster. So the topology was for 9, but the cluster was for 3.
Maybe we can change the naming a bit?

Also, as Tomer pointed out to me recently, this doesn't cover all of the use cases, as sometimes you need devices 4-6 and not 1-3 (which is what is given here).

Any suggestions on the syntax for that?

Reply:
I see, thanks!
Then maybe something like subsets=(4, 5, 6) could work no? Like this you can also define a subset as non-neighbors like (1, 5, 6) for instance, although I don't know if it makes sense...

Reply:
Done